core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -2004,7 +2004,8 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext
 
     // executor1 is eventually lost
     scheduler.executorLost("executor1", ExecutorExited(0, false, "normal"))
-    assert(scheduler.executorsPendingDecommission.isEmpty)
+    // [SPARK-40481] after executor lost, decommission state still be kept
+    assert(scheduler.getExecutorDecommissionState("executor1").isDefined)
```

Comment:
Also validate the cleanup behavior.
For example, set size to 2 and decom 3 executors to validate first got evicted - ensures size is bounded as code evovles.